src/bundle.c

```diff
@@ -505,7 +505,7 @@ static gboolean convert_to_casync_bundle(RaucBundle *bundle, const gchar *outbun
 	g_return_val_if_fail(outbundle, FALSE);
 	g_return_val_if_fail(error == NULL || *error == NULL, FALSE);
 
-	basepath = g_strndup(outbundle, strlen(outbundle) - 6);
+	basepath = g_strndup(outbundle, strlen(outbundle) > 6 ? strlen(outbundle) - 6 : 0);
```

Comment:
strlen()'s second argument can be rearranged to be `strlen(outbundle) - 6 > 0 ? strlen(outbundle) - 6 : 0` revealing that it can be rearranged again to be `MAX(strlen(outbundle) - 6, 0)` which is readable more easily.
The number `6` presumably is the result of `strlen(".raucb")`, right? Hence I suggest to finally use the expression `MAX(strlen(outbundle) - strlen(".raucb"), 0)` to further enhance the code's understandability.
By the way: there is no check anywhere that `outbundle` really ends in ".raucb" which presumably is why you started this pull request. What about ensuring that instead of the current approach?

Reply:
Thanks for your suggestions. It is not a requirement anymore that bundles have to end with `.raucb`. Thus we cannot check for this anymore.
Maybe we need a cleaner solution than my hack. It is a result of testing and I mainly did not want it to get lost ;)